File: fapi/src/main/java/com/netcracker/edu/fapi/models/Post.java
package com.netcracker.edu.fapi.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import java.sql.Timestamp;
import java.util.List;
import java.util.Set;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Post {
    private Long id;
    private String photo;
    private String txt;
    private Timestamp date;
    private String place;
    private User user;
    private List<Comment> comment;
    private List<Tag> tags;
    private Set<User> likes;
}

File: frontend/src/app/moduls/post.ts
import {User} from './user';
import {Comments} from './comments';
import {Tag} from './tag';
import {Like} from './like';

export class Post{

  id: number;
  photo: string;
  txt: string;
  date: Date;
  place: string;
  user: User;
  comment: Comments[];
  tags: Tag[];
  likes: Like[];

  constructor() {
  }
}
